runtime/src/tiered_storage/byte_block.rs

```diff
@@ -30,13 +36,46 @@ impl ByteBlockWriter {
     pub fn write(&mut self, buf: &[u8]) -> std::io::Result<()> {
         match self {
             Self::Raw(cursor) => cursor.write_all(buf)?,
+            Self::Lz4(lz4_encoder) => lz4_encoder.write_all(buf)?,
         };
         Ok(())
     }
 
     pub fn finish(self) -> std::io::Result<Vec<u8>> {
         match self {
             Self::Raw(cursor) => Ok(cursor.into_inner()),
+            Self::Lz4(lz4_encoder) => {
+                let (compressed_block, result) = lz4_encoder.finish();
+                result?;
```

Comment:
`lz4::Encoder` returns `(W, Result<()>)`.  I am wondering whether there is a cleaner way to write the above code 🤔 .  Tried `let (compressed_block, result?) = lz4_encoder.finish();` and `lz4_encoder.finish()?` but none of them works.